zipper/protocols/irondb/irondb_group.go

```diff
@@ -0,0 +1,682 @@
+				// fallback to IRONdb Fetch API
+				query = graphiteExprListToIronDBTagQuery(strings.Split(query, ","))
+				findTagOptions := &gosnowth.FindTagsOptions{
+					// start and stop according to request
+					Start:     time.Unix(start, 0),
+					End:       time.Unix(stop, 0),
+					Activity:  0,
+					Latest:    0,
+					CountOnly: 0,
+					Limit:     -1,
+				}
+				logger.Debug("send tag find result to irondb",
+					zap.String("query", query),
+					zap.Any("findTagOptions", findTagOptions),
+				)
+				stats.FindRequests++
+				tagMetrics, err := c.client.FindTags(c.accountID, query, findTagOptions)
+				if err != nil {
+					e = processFindErrors(err, e, stats, query)
+					continue
+				}
+				logger.Debug("got tag find result from irondb",
+					zap.String("query", query),
+					zap.Any("tagMetrics", tagMetrics),
+				)
+				responses := []*gosnowth.DF4Response{}
+				for _, metric := range tagMetrics.Items {
+					stats.RenderRequests++
+					res, err2 := c.client.FetchValues(&gosnowth.FetchQuery{
+						Start:  time.Unix(start, 0),
+						Period: time.Duration(step) * time.Second,
+						Count:  count,
+						Streams: []gosnowth.FetchStream{{
+							UUID:      metric.UUID,
+							Name:      metric.MetricName,
+							Kind:      metric.Type,
+							Label:     metric.MetricName,
+							Transform: "average",
+						}},
+						Reduce: []gosnowth.FetchReduce{{
+							Label:  "pass",
+							Method: "pass",
+						}},
+					})
+					if err2 != nil {
+						e = processRenderErrors(err2, e, stats, query)
+						continue
+					}
+					responses = append(responses, res)
+				}
+				logger.Debug("got fetch result from irondb",
+					zap.String("query", query),
+					zap.Any("responses", responses),
+				)
+				for _, response := range responses {
+					// We always should trust backend's response (to mimic behavior of graphite for grahpite native protoocols)
+					// See https://github.com/go-graphite/carbonapi/issues/504 and https://github.com/go-graphite/carbonapi/issues/514
+					realStart := start
+					realStop := stop
+					if len(response.Data) > 0 {
+						realStart = response.Head.Start
+						realStop = response.Head.Start + response.Head.Period*(response.Head.Count-1)
+					}
+					for i, meta := range response.Meta {
+						values := make([]float64, (realStop-realStart)/step+1)
+						for _, data := range response.Data[i] {
+							dv, ok := data.(float64)
+							if ok {
+								values = append(values, dv)
+							} else {
+								values = append(values, math.NaN())
+							}
+						}
+						name := convertNameToGraphite(meta.Label)
+						r.Metrics = append(r.Metrics, protov3.FetchResponse{
+							Name:              name,
+							PathExpression:    pathExpr,
+							ConsolidationFunc: "Average",
+							StartTime:         realStart,
+							StopTime:          realStop,
+							StepTime:          step,
+							Values:            values,
+							XFilesFactor:      0.0,
+						})
+					}
+				}
```

Comment:
This is a personal preference, but I'd rather use `continue` here and get rid of `else` for reducing the nesting level.